### PR TITLE
use sha256 checksum instead of md5

### LINKED ISSUE #29389: Security: Downloaded plugin zip files should be verified with sha256 instead of md5.
`G401 (CWE-326):  Use of weak cryptographic primitive`
 `pkg/cmd/grafana-cli/services/api_client.go:97`

The checksum comes from the grafana.com API so we would have to verify that it has a sha256 checksum of the plugin as well.

### PR DESCRIPTION
**Pull request by ying-jeanne**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #29389 

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by ying-jeanne** [comment]
I still need to test manually the plugin which is not downloaded from github, the checksum works correctly.

File: pkg/cmd/grafana-cli/services/api_client.go [hunk 1]
```diff
@@ -101,15 +101,15 @@ func (client *GrafanaComClient) DownloadFile(pluginName string, tmpFile *os.File
 	}()
 
 	w := bufio.NewWriter(tmpFile)
-	h := md5.New()
+	h := sha256.New()
 	if _, err = io.Copy(w, io.TeeReader(bodyReader, h)); err != nil {
-		return errutil.Wrap("Failed to compute MD5 checksum", err)
+		return errutil.Wrap("Failed to compute sha256 checksum", err)
```
Comment (aknuds1) [inline pkg/cmd/grafana-cli/services/api_client.go:106]: Go errors should generally start with a lowercase letter:
```suggestion
		return errutil.Wrap("failed to compute SHA256 checksum", err)
```

File: pkg/cmd/grafana-cli/services/api_client.go [hunk 2]
```diff
@@ -101,15 +101,15 @@ func (client *GrafanaComClient) DownloadFile(pluginName string, tmpFile *os.File
...
 	}
 	if err := w.Flush(); err != nil {
 		return fmt.Errorf("failed to write to %q: %w", tmpFile.Name(), err)
 	}
 	if len(checksum) > 0 && checksum != fmt.Sprintf("%x", h.Sum(nil)) {
-		return fmt.Errorf("expected MD5 checksum does not match the downloaded archive - please contact security@grafana.com")
+		return fmt.Errorf("expected sha256 checksum does not match the downloaded archive - please contact security@grafana.com")
```
Comment (aknuds1) [inline pkg/cmd/grafana-cli/services/api_client.go:112]:
```suggestion
		return fmt.Errorf("expected SHA256 checksum does not match the downloaded archive - please contact security@grafana.com")
```

**Review by aknuds1 — COMMENTED** [review]
Generally LGTM, but see suggestions.

**Review by aknuds1 — CHANGES_REQUESTED** [review]
I think we can remove MD5 references in grafana-cli now, want me to push a commit for this? Should also fix the test failure.

**Review by aknuds1 — APPROVED** [review]
LGTM

**Pull request merged**